src/main/java/com/zendesk/maxwell/producer/partitioners/MaxwellKafkaPartitioner.java

```diff
@@ -26,11 +33,17 @@ public MaxwellKafkaPartitioner(String hashFunction, String partitionKey) {
 				break;
 			case "primary_key": this.provider = new HashStringPrimaryKey();
 				break;
+			case "column": this.provider = new HashStringColumn();
+				break;
 			case "database":
 			default:
 				this.provider = new HashStringDatabase();
 				break;
 		}
+		if ( partitionColumns != null )
+			this.partitionColumns = Arrays.asList(csvColumns.split(","));
```

Comment:
`partitionColumns` will never be null -- you're doing the null check on the wrong variable